stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListItemStyle.kt

```diff
@@ -0,0 +1,97 @@
+        @ColorInt
+        private var messageBackgroundColorMine: Int = 0
+
+        @ColorInt
+        private var messageBackgroundColorTheirs: Int = 0
+
+        @ColorInt
+        private var messageTextColorMine: Int = 0
+
+        @ColorInt
+        private var messageTextColorTheirs: Int = 0
+
+        @ColorInt
+        private var messageLinkTextColorMine: Int = 0
+
+        @ColorInt
+        private var messageLinkTextColorTheirs: Int = 0
+
+        fun messageBackgroundColorMine(
+            @StyleableRes messageBackgroundColorMineStyleableId: Int,
+            @ColorInt defaultValue: Int = VALUE_NOT_SET,
+        ) = apply {
+            messageBackgroundColorMine = attributes.getColor(messageBackgroundColorMineStyleableId, defaultValue)
+        }
+
+        fun messageBackgroundColorTheirs(
+            @StyleableRes messageBackgroundColorTheirsId: Int,
+            @ColorInt defaultValue: Int = VALUE_NOT_SET,
+        ) = apply {
+            messageBackgroundColorTheirs = attributes.getColor(messageBackgroundColorTheirsId, defaultValue)
+        }
+
+        fun messageTextColorMine(
+            @StyleableRes messageTextColorMineId: Int,
+            @ColorInt defaultValue: Int = VALUE_NOT_SET,
+        ) = apply {
+            messageTextColorMine = attributes.getColor(messageTextColorMineId, defaultValue)
+        }
+
+        fun messageTextColorTheirs(
+            @StyleableRes messageTextColorTheirsId: Int,
+            @ColorInt defaultValue: Int = VALUE_NOT_SET,
+        ) = apply {
+            messageTextColorTheirs = attributes.getColor(messageTextColorTheirsId, defaultValue)
+        }
+
+        fun messageLinkTextColorMine(
+            @StyleableRes messageLinkTextColorMineId: Int,
+            @ColorInt defaultValue: Int = VALUE_NOT_SET,
+        ) = apply {
+            messageLinkTextColorMine = attributes.getColor(messageLinkTextColorMineId, defaultValue)
+        }
+
+        fun messageLinkTextColorTheirs(
+            @StyleableRes messageLinkTextColorTheirsId: Int,
+            @ColorInt defaultValue: Int = VALUE_NOT_SET,
+        ) = apply {
+            messageLinkTextColorTheirs = attributes.getColor(messageLinkTextColorTheirsId, defaultValue)
+        }
+
+        fun build(): MessageListItemStyle {
+            return MessageListItemStyle(
+                messageBackgroundColorMine = messageBackgroundColorMine.nullIfNotSet(),
+                messageBackgroundColorTheirs = messageBackgroundColorTheirs.nullIfNotSet(),
+                messageTextColorMine = messageTextColorMine.nullIfNotSet(),
+                messageTextColorTheirs = messageTextColorTheirs.nullIfNotSet(),
+                messageLinkTextColorMine = messageLinkTextColorMine.nullIfNotSet(),
+                messageLinkTextColorTheirs = messageLinkTextColorTheirs.nullIfNotSet(),
+            )
+        }
+
+        private fun Int.nullIfNotSet(): Int? {
+            return if (this == VALUE_NOT_SET) null else this
+        }
```

Comment:
Really neat solution!

Reply:
Thanks! I was wondering if I made this more complicated than it needed to be, but it was the only way I could figure to do it until (if) we support all the other attributes.